### ISSUE TITLE
Priority tag on completed tasks breaks filtering, rendering

### ISSUE THREAD

**Issue by Serneum** [mvgrimes/obsidian-todotxt-plugin#16]
This may be intentional, but I've noticed that the priority of a ToDo item is saved as a tag `pri:`, which seems to cause issues with modifying the item after it has been completed. For example, if I complete an item but then want to change the context or project, the tag shows up and then when saved, it does not render with the little priority bubble. When modifying a ToDo item that has not been completed, this does not seem to be an issue at all.

This also occurs when using `todo.sh archive` later:
![image](https://github.com/mvgrimes/obsidian-todotxt-plugin/assets/1091716/e7f4072d-0049-4ddf-932d-da1c14cf0b90)

This causes the priority filter to not work as expected and I need to set it to `All` to see the tasks that were completed. Maybe that's okay, it just struck me as odd

**Comment by Serneum** [comment]
It looks like this may be intentional per [the code](https://github.com/mvgrimes/obsidian-todotxt-plugin/blob/fcebc23fa2b74a87cd3ec8842bee2c83ecb4008e/src/lib/todo.ts#L67) and [todo.txt syntax](https://github.com/todotxt/todo.txt#rule-2-the-date-of-completion-appears-directly-after-the-x-separated-by-a-space) so I will close this.

**Comment by mvgrimes** [comment]
Yes this is intentional, but it really was just part of an experimental feature that I'll be releasing in 1.3.4: when the "Preserve Priority" feature is enabled, tasks that are toggled to *not completed* will have their priority restored from the `pri:` tag.

As for displaying tasks, if a task had a priority equal to or higher than the priority filter, it is shown for a time after being marked complete. Once the file has been reloaded, completed tasks are hidden regardless of their (former) priority.
